backend/testing/messages/test_object_schema.py

```diff
@@ -0,0 +1,109 @@
+import unittest
+
+from messages.object_schema import ObjectSchema
+from data.management.user import User
+from business_objects.business_attribute_base import BaseFlag
+from business_objects.business_object_base import BOBase
+from business_objects.bo_descriptors import (
+    AttributeAccessLevel,
+    AttributeType,
+    BOFlag,
+    BOStr,
+    BOList,
+    BORelation,
+)
+
+
+class OtherMockBO(BOBase):
+    pass
+
+
+class MockBOFlag(BaseFlag):
+    FLAG_1 = 1
+    FLAG_2 = 2
+
+
+class MockBO(BOBase):
+    def __new__(cls, id: int | None = None, *args, **attributes):
+        print(f"MockBOBase.__new__({cls=}, {id=}, {args=}, {attributes=})")
+        return super().__new__(cls)
+
+    # Really should mock attribute_descriptions instead...
+    mock_str = BOStr()
+    mock_list = BOList()
+    mock_relation = BORelation(OtherMockBO)
+    mock_flag = BOFlag(MockBOFlag)
+
+
+class MockAttributeDescription:
+    def __init__(
+        self,
+        name,
+        attribute_type,
+        flag_values,
+        access_level=AttributeAccessLevel.AAL_READ_WRITE,
+    ):
+        self.name = name
+        self.attribute_type = attribute_type
+        self.flag_values = flag_values
+        self.access_level = access_level
+
+
+class Test_100__ObjectSchema(unittest.TestCase):
+
+    def bo_payload_test(self, obj_type: type[BOBase]):
+        obj = ObjectSchema(object_type=obj_type)
+        payload = obj.generate_payload()
+        for v in obj_type.attribute_descriptions():
+            if v.access_level == AttributeAccessLevel.AAL_WRITE_ONLY:
+                self.assertNotIn(v.name, payload)
+                continue
+            self.assertEqual(
+                {
+                    "type": obj.attribute_type_representation(v.attribute_type),
+                    "flags": {
+                        k: (obj.flag_values_representation(val))
+                        for k, val in v.flag_values.items()
+                    },
+                },
+                payload[v.name],
+            )
+
+    def test101_initialization(self):
```

Comment:
```suggestion
    def test_101_initialization(self):
```

Test methods are listed alphabetically in VSC. The name should always begin with 'test_' followed by the number.